src/prometheus.ml

```diff
@@ -236,8 +251,150 @@ module Summary = struct
     let start = gettimeofday () in
     Lwt.finalize fn
       (fun () ->
-         let finish = Unix.gettimeofday () in
+         let finish = gettimeofday () in
          observe t (finish -. start);
          Lwt.return_unit
       )
 end
+
+type histogram = {
+    upper_bounds: float array;
+    counts: float array;
+    mutable sum: float;
+  }
+
+let _histogram at_index_f count =
+  let real_at_index i =
+    if i >= count then
+      infinity
+    else
+      at_index_f i
+  in
+  let upper_bounds = (Array.create_float (count + 1)) in
```

Comment:
I think you can just use `Array.init` here and avoid a lot of complexity, and continue to work with all OCaml versions.